src/main/java/core/basesyntax/service/FileWriter.java

```
@@ -0,0 +1,7 @@
+package core.basesyntax.service;
+
+import java.io.IOException;
+
+public interface FileWriter {
+    void write(String data, String filePath) throws IOException;
```

This method declares `throws IOException` which is inconsistent with FileWriterImpl that catches IOException and rethrows a RuntimeException with an informative message. Either remove the checked exception from the interface (and update callers) or let the implementation propagate IOException. The checklist recommends rethrowing informative RuntimeException in catch blocks, so if you keep that approach, update the interface signature accordingly.